tests/unit/handlers/test_handlers.py

```diff
@@ -154,6 +158,47 @@ def test_record_with_request(self):
             self.assertEqual(record._trace_str, expected_trace)
             self.assertEqual(record._span_id, expected_span)
             self.assertEqual(record._span_id_str, expected_span)
+            self.assertTrue(record._trace_sampled)
+            self.assertEqual(record._trace_sampled_str, "true")
+            self.assertEqual(record._http_request, expected_request)
+            self.assertEqual(record._http_request_str, json.dumps(expected_request))
+
+    def test_record_with_traceparent_request(self):
+        """
+        test filter adds http request data when available
+        """
+        import logging
+
+        filter_obj = self._make_one()
+        record = logging.LogRecord(None, logging.INFO, None, None, None, None, None,)
+        record.created = None
+
+        expected_path = "http://testserver/123"
+        expected_agent = "Mozilla/5.0"
+        expected_trace = "4bf92f3577b34da6a3ce929d0e0e4736"
+        expected_span = "00f067aa0ba902b7"
+        combined_trace = f"00-{expected_trace}-{expected_span}-03"
```

Comment:
nit: you might want to set the flags field to `01` to keep aligned with [other flags](https://www.w3.org/TR/trace-context/#other-flags) of the standard.
on the same account would you like to add sampling=false test?

Reply:
trace_sampled should be True if the last bit in the flags byte string is True. I wanted to test the more complex case of "0...011" instead of the simple case of `0...001`. (The 01 case and others are thoroughly tested elsewhere in test__helpers. test_handlers is slightly more high level)